rotkehlchen/tests/unit/test_chains_aggregator.py

```diff
@@ -26,42 +27,81 @@ def test_module_deactivation(blockchain):
-            optimism_addresses=[optimism_addy, all_addy],
+            eth_contract_addresses=[eth_addy_contract],
+            ethereum_addresses=[eth_addy_contract, addy_eoa],
+            optimism_addresses=[addy_eoa],
+            avalanche_addresses=[addy_eoa],
         )
-        assert set(blockchain.filter_active_evm_addresses([contract_addy])) == {
-            (SupportedBlockchain.ETHEREUM, contract_addy),
-        }
-        assert set(blockchain.filter_active_evm_addresses([all_addy])) == {
-            (SupportedBlockchain.ETHEREUM, all_addy),
-            (SupportedBlockchain.AVALANCHE, all_addy),
-            (SupportedBlockchain.OPTIMISM, all_addy),
-        }
-        assert set(blockchain.filter_active_evm_addresses([avalanche_addy])) == {
-            (SupportedBlockchain.ETHEREUM, avalanche_addy),
-            (SupportedBlockchain.AVALANCHE, avalanche_addy),
-        }
-        assert set(blockchain.filter_active_evm_addresses([optimism_addy])) == {
-            (SupportedBlockchain.ETHEREUM, optimism_addy),
-            (SupportedBlockchain.OPTIMISM, optimism_addy),
-        }
-        assert set(blockchain.filter_active_evm_addresses([optimism_addy, all_addy, contract_addy])) == {  # noqa: E501
-            (SupportedBlockchain.ETHEREUM, contract_addy),
-            (SupportedBlockchain.ETHEREUM, all_addy),
-            (SupportedBlockchain.AVALANCHE, all_addy),
-            (SupportedBlockchain.OPTIMISM, all_addy),
-            (SupportedBlockchain.ETHEREUM, optimism_addy),
-            (SupportedBlockchain.OPTIMISM, optimism_addy),
-        }
+
+        blockchain.detect_evm_accounts()
+
+    assert set(blockchain.accounts.eth) == {addy_eoa, eth_addy_contract}
+    assert blockchain.accounts.optimism == [addy_eoa]
+    assert blockchain.accounts.avax == [addy_eoa]
+
+    # Also check the db
+    expected_accounts_data = initial_accounts_data + [
+        BlockchainAccountData(
+            chain=SupportedBlockchain.ETHEREUM,
+            address=addy_eoa,
+        ),
+        BlockchainAccountData(
+            chain=SupportedBlockchain.AVALANCHE,
+            address=addy_eoa,
+        ),
+    ]
+    accounts_in_db = []
+    with blockchain.database.conn.read_ctx() as cursor:
+        raw_accounts = blockchain.database.get_blockchain_accounts(cursor)
+        for account in raw_accounts.eth:
+            accounts_in_db.append(BlockchainAccountData(
+                chain=SupportedBlockchain.ETHEREUM,
+                address=account,
+            ))
+        for account in raw_accounts.optimism:
+            accounts_in_db.append(BlockchainAccountData(
+                chain=SupportedBlockchain.OPTIMISM,
+                address=account,
+            ))
+        for account in raw_accounts.avax:
+            accounts_in_db.append(BlockchainAccountData(
+                chain=SupportedBlockchain.AVALANCHE,
+                address=account,
+            ))
+
+    assert set(accounts_in_db) == set(expected_accounts_data)
```

Comment:
is the set needed? The db result should be always in the same order right? if it is because they are mixed can you assert that there is the same number of results and set didn't remove any duplicate that might have been introduced?

Reply:
They are mixed because one is all initial addresses and then the ones that were detected, while the other is eth addresses, op addresses, avax addresses. Will add a length assertion